# Investment Multiple and CAGR (Compound Annual Growth Rate)

### Two alternative reward metrics that are more intuitive and easier to interpret (compared to the arithmetic mean).

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
pd.options.display.float_format = '{:.4f}'.format
plt.style.use("seaborn")

In [2]:
msft = pd.read_csv("../../Assets/Data-Files/msft.csv", index_col = "Date", parse_dates = ["Date"])
msft

Price  Returns
Date                        
2014-09-30  46.3600      NaN
2014-10-01  45.9000  -0.0099
2014-10-02  45.7600  -0.0031
2014-10-03  46.0900   0.0072
2014-10-06  46.0900   0.0000
...             ...      ...
2021-05-24 250.7800   0.0229
2021-05-25 251.7200   0.0037
2021-05-26 251.4900  -0.0009
2021-05-27 249.3100  -0.0087
2021-05-28 249.6800   0.0015

[1678 rows x 2 columns]

**Investment Multiple:** Ending Value of 1 (Dollar, Euro, etc.) invested.  
Multiple = Ending Value / Initial Value

In [5]:
# last price of a given period divided by the first price returns the multiple
multiple = (msft.Price[-1] / msft.Price[0])
multiple # result is the ending value of one dollar invested

5.3856770791334085

**Price Increase (in %)**

In [7]:
# calculate the increase in percent
(multiple - 1) * 100 # the invested dollar gets subtracted from the multiple

438.5677079133409

In [9]:
# similar (not identical) concept: Normalized Price with a Base Value 1
msft.Price / msft.Price[0] # last value is equivalent to multiple

Date
2014-09-30   1.0000
2014-10-01   0.9901
2014-10-02   0.9871
2014-10-03   0.9942
2014-10-06   0.9942
              ...  
2021-05-24   5.4094
2021-05-25   5.4297
2021-05-26   5.4247
2021-05-27   5.3777
2021-05-28   5.3857
Name: Price, Length: 1678, dtype: float64

**Drawback of Investment Multiple:** Doesn't take into account investment period (5.38 is great over seven years but not so great over 100 years). Meaningful only in conjunction with investment period (this means investment period has to be taken into account which is the benefit of the CAGR).

<font size="4" style="line-height:1.3">**Compound Annual Growth Rate (CAGR):** The (constant annual) rate of return that would be required for an investment to grow from its beginning balance to its ending balance, assuming the profits were reinvested at the end of each year of the investment's lifespan (Wikipedia).</font>

In [13]:
start = msft.index[0]
start

Timestamp('2014-09-30 00:00:00')

In [12]:
end = msft.index[-1]
end

Timestamp('2021-05-28 00:00:00')

In [17]:
# timestamp objects can be subtracted
td = end - start
td # returns a timedelta object

Timedelta('2432 days 00:00:00')

In [15]:
# timedelta divided by average number of years = 365.25
td_years = td.days / 365.25
td_years

6.658453114305271

In [19]:
# Compound Annual Growth Rate is the multiple to the power of one devided by the annual timedelta
# and then subtracted by one
cagr = multiple**(1 / td_years) - 1 #short version
cagr

0.2877197304182786

In [27]:
# more complex calculation with pandas data frame
# (msft.Price[-1]/msft.Price[0]) calculates the multiple
# then the timedelta divided by average number of years = 365.25 minus 1 is calculated
cagr = (msft.Price[-1]/msft.Price[0])**(1/((msft.index[-1] - msft.index[0]).days / 365.25)) - 1 
cagr

0.2877197304182786

In [28]:
# inverse calculation, calculating multiple with cagr
(1 + cagr)**(td_years) # alternative 2 to calculate multiple

5.38567707913341

**Take Home:** CAGR can be used to compare investments with different investment horizons.